lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -1,124 +1,168 @@
-      attr_reader :api_url, :ingress_url
+      attr_reader :logger
 
-      def initialize(api_url, ingress_url, collector_definitions_file)
-        raise "Access to the cluster using serviceaccounts is required" unless ObjectManager.available?
-        @api_url               = api_url
-        @ingress_url           = ingress_url
-        @object_manager        = ObjectManager.new
-        @collector_definitions = YAML.load_file(collector_definitions_file)
+      def initialize(api_base_url: ENV["API_URL"], collector_definitions_file: ENV["CDF"])
+        @api_base_url = api_base_url
+        @collector_definitions_file = collector_definitions_file || TopologicalInventory::Orchestrator.root.join("config/collector_definitions.yaml")
+        @logger = ManageIQ::Loggers::Container.new
       end
 
       def run
         loop do
-          Signal.trap("TERM") { break }
-          collector_definitions.each do |source_type, endpoint_info|
-            endpoint_info.each do |endpoint_role, options|
-              resolve_collector_type(source_type, endpoint_role, options)
-            end
-          end
+          make_openshift_match_database
+
           sleep 10
         end
       end
 
-      def resolve_collector_type(source_type, endpoint_role, options)
+      private
+
+      def digest(object)
+        require 'digest'
+        Digest::SHA1.hexdigest(Marshal.dump(object))
       end
 
-      def create_collectors_for_new_sources
-        sources.each do |source|
-          create_objects_for_source(source) unless have_deployment_for_source?(source)
-        end
+      def make_openshift_match_database
+        hash = {}
+        expected = collectors_from_database(hash)
+        current  = collectors_from_openshift
+
+        logger.info("Checking...")
+
+        (current - expected).each { |i| remove_openshift_objects_for_source(i) }
+        (expected - current).each { |i| create_openshift_objects_for_source(i, hash[i]) }
+
+        logger.info("Checking... complete.")
       end
-      
-      def remove_collectors_for_deleted_sources
-        collector_deployments.each do |deployment|
-          unless sources.include?(source_for_deployment(deployment))
-            @object_manager.delete_deployment_config(deployment.metadata.name)
-            @object_manager.delete_secret("#{deployment.metadata.name}-secrets")
+
+
+      ### API STUFF
+      def each_endpoint
+        return enum_for(:each_endpoint) unless block_given?
+        each_resource(url_for("source_types")) do |source_type|
+          collector_definition = collector_definitions[source_type["name"]]
+          next if collector_definition.nil?
+          each_resource(url_for("source_types/#{source_type["id"]}/sources")) do |source|
+            each_resource(url_for("sources/#{source["id"]}/endpoints")) do |endpoint|
+              each_resource(url_for("authentications?resource_type=Endpoint&resource_id=#{endpoint["id"]}&authtype=default")) do |authentication|
```

Comment:
What are the possible values for `authtype`? Right now we have it as `nil`, but will it always be `default` or should we just take the first one?

Reply:
@agrare ?

Reply:
This is supposed to be if this was e.g. a username+password/ssh token/etc.. if providers support multiple types of creds.

Having a column that represents if the authentication is default for the endpoint would be useful though if we end up in a situation where we need to store multiple user credentials for the same endpoint for e.g. running operations as a user on a provider without impersonation.

Reply:
It sounds like we would need a validation around this to verify that there is only one default for the endpoint.

Reply:
>where we need to store multiple user credentials for the same endpoint

This may make it more difficult to transition away from the authentications table when the replacement service(s) become available

Reply:
We would still have our own auth records for metadata just would fetch the password and maybe username from the vault service.